packages/fx-core/src/core/middleware/projectMigrator.ts

```diff
@@ -999,45 +1023,187 @@ async function updateConfig(ctx: CoreHookContext) {
   await fs.writeFile(path.join(fx, "new.env.default.json"), JSON.stringify(envConfig, null, 4));
 }
 
+export async function generateBicepsV3(
+  projectSettings: ProjectSettingsV3,
+  inputs: InputsWithProjectPath
+): Promise<Result<undefined, FxError>> {
+  //init bicep folder
+  {
+    const bicepComponent = Container.get<BicepComponent>("bicep");
+    const res = await bicepComponent.init(inputs.projectPath!);
+    if (res.isErr()) return err(res.error);
+  }
+  const biceps: Bicep[] = [];
+  const context = createContextV3(projectSettings);
+  // teams-tab
+  {
+    const config = getComponent(projectSettings, ComponentNames.TeamsTab);
+    if (config) {
+      const hosting = config.hosting! || ComponentNames.AzureStorage;
```

Comment:
also just for learning. Why we do "|| ComponentNames.AzureStorage" here?

Reply:
For case when there is no hosting, actually, it may not happen if there is no bug for project settings migration.